caom2utils/caom2utils/fits2caom2.py

```diff
@@ -2175,8 +2314,22 @@ def _get_members(self, obs):
                 ('Cannot apply blueprint for CompositeObservation to a '
                  'simple observation'))
         elif isinstance(obs, CompositeObservation):
-            members = self._get_from_list('CompositeObservation.members',
-                                          index=0)
+            lookup = self.blueprint._get('CompositeObservation.members',
+                                         extension=1)
+            if ObsBlueprint.is_table(lookup) and len(self.headers) > 1:
+                member_list = self._get_from_table(
+                    'CompositeObservation.members', 1)
+                # ensure the members are good little ObservationURIs
+                if member_list.startswith('caom:'):
+                    members = member_list
+                else:
+                    members = ''
+                    for ii in member_list.split():
+                        members = 'caom:{}/{} {}'.format(
+                            obs.collection, ii, members)
```

Comment:
An equivalent one liner that checks and fixes every entry using list comprehensions:
' '.join(['caom:{}'.format(i) if not i.startswith('caom:') else i for i in member_list.split()])

e.g.:
' '.join(['caom:{}'.format(i) if not i.startswith('caom:') else i for i in 'a caom:b c'.split()])

Reply:
Done.